nixos/modules/services/networking/netmaker/netmaker.nix

```diff
@@ -0,0 +1,787 @@
+      '';
+    };
+
+    telemetry = lib.mkEnableOption "sending telemetry to Netmaker developers";
+
+    environment = lib.mkOption {
+      type = with lib.types; attrsOf str;
+      default = { };
+      apply = envAttrs: pkgs.writeText "environment-env" (lib.trivial.pipe envAttrs [
+        (lib.attrsets.mapAttrsToList (name: value: "${name}=${lib.strings.escapeShellArg value}"))
+        (builtins.concatStringsSep "\n")
+      ]);
+      description = lib.mdDoc ''
+        Additional environment variables to set on Netmaker Server
+      '';
+    };
+
+    environmentFiles = lib.mkOption {
+      type = with lib.types; listOf path;
+      default = [ ];
+      description = lib.mdDoc ''
+        Additional files holding environment variable definitions to set on Netmaker Server
+      '';
+    };
+
+    config = lib.mkOption {
+      type = with lib.types; anything;
+      description = lib.mdDoc ''
+        Content of Netmaker configuration file.
+
+        see for more information:
+        - https://github.com/gravitl/netmaker/blob/dc8f9b1bc74d262669bf96fab5f0e545f5906432/config/config.go
+      '';
+      apply = value: pkgs.writeText "netmaker-config.json" (builtins.toJSON value);
+    };
+
+    installType = lib.mkOption {
+      type = with lib.types; enum [
+        "ce"
+        "pro"
+      ];
+      default = "ce";
+      description = lib.mdDoc ''
+        Type of the installation: `ce` for Community and `pro` for Professional.
+
+        `pro` requires additional configuration not covered by the module, for reference see the [code search](https://github.com/search?q=repo%3Agravitl%2Fnetmaker+%22%24INSTALL_TYPE%22&type=code)
+      '';
+    };
+
+    dataDir = lib.mkOption {
+      type = with lib.types; path;
+      default = "/var/lib/netmaker";
+      description = lib.mdDoc ''
+        Netmaker data directory, currently simply a working directory the server is running in.
+        It contains following known directories and files:
+        - `data/netmaker.db` - sqlite database file
+        - `data/netmaker.log.YYYY-MM-DD` - Netmaker server log files
+        - `config/dnsconfig/netmaker.hosts` - `/etc/hosts` formatted file dynamically updated by Netmaker
+        - `config/dnsconfig/Corefile` - generated by Netmaker, but not used in this module
+
+        Current list of files can be found in [this GitHub search](https://github.com/search?q=repo%3Agravitl%2Fnetmaker+%22os.Getwd%28%29%22&type=code)
+      '';
+    };
+
+    configDir = lib.mkOption {
+      type = with lib.types; path;
+      default = "/etc/netmaker";
+      description = lib.mdDoc ''
+        A configuration directory to (currently) put NixOS module related configuration files in.
+
+        Following configuration files are used by default:
+        - `''${config.services.netmaker.configDir}/env` - user editable environment variables
+        - `''${config.services.netmaker.configDir}/env.generated` - generated environment variables
```

Comment:
editable data belongs under /var/lib. /etc/ only really holds hardcoded configs which we cannot really pass directly to the program or we need a static path for other reasons.